### PR TITLE
impute each row works with ~ .8 corr coeff

### PR DESCRIPTION
**Pull request by Farnazmdi**
Not sure what I was doing wrong, but I ran it again with cross-validating each row at a time, and it works. correlation coefficient around 0.8 for 20 iterations. Would you like a plot for this?

### CODE REVIEW

**Review by aarmey — CHANGES_REQUESTED** [review]
Great news! How do you think a plot would look?

File: de/impute.py [hunk 1]
```diff
@@ -7,16 +7,15 @@
 from .fancyimpute.soft_impute import SoftImpute
 
 
-def split_data(X, n=20):
```
[the remark this answers is not in the document]
Comment (Farnazmdi) [reply de/impute.py:10]: I made it 1 in 30 data points.. It was not being passed as an argument to the `repeatImpute`. I will bring it back.. now that I think, it depends on the data size.

File: de/impute.py [hunk 2]
```diff
@@ -7,16 +7,15 @@
...
-    row = np.random.choice(X.shape[0], n, replace=False)
-    col = np.random.choice(X.shape[1], n, replace=False)
+    row = np.random.choice(X.shape[0], 1, replace=False)
+    col = np.random.choice(X.shape[1], np.int(X.shape[1] / 30), replace=False)
     train_X = np.copy(X)
     test_X = np.full_like(X, np.nan)
     train_X[row, col] = np.nan
     test_X[row, col] = X[row, col]
-    assert np.sum(np.isnan(train_X)) == n
-    assert np.sum(np.isfinite(test_X)) == n
```
[the remark this answers is not in the document]
Comment (Farnazmdi) [reply de/impute.py:19]: I will bring it back.. I removed it since I had removed `n`.

**Comment by Farnazmdi** [comment]
For the plot, maybe I can plot the correlation versus the number of perturbations we pick from each gene, which could also help find the number of randomly chosen perturbations (n) to minimize prediction error?

Another way, could be a boxplot of the correlation coefficients for many iterations that each time we pick a random gene, with a fixed number of random perturbations per gene.

**Comment by aarmey** [comment]
I don't understand your description here.

**Comment by Farnazmdi** [comment]
Here is another suggestion which I think is better:
a boxplot for the correlation coefficients of imputation, across cell lines, comparing the linear and nonlinear model.

**Comment by aarmey** [comment]
I don't think there's a need to visualize anything at this point... I suppose comparing to the linear model would be fine.

**Comment by Farnazmdi** [comment]
I wrote a function anyway, we can just keep it around and use it when the time arrived. Do you agree?
It should plot something like this:

image

**Comment by aarmey** [comment]
Yes, that's fine.

I can't go back and forth on these things all day. Please collect these sort of questions so that we can address them when we meet.

**Pull request merged**